### ISSUE TITLE
Add exmplars into metrics generated for ingesting traces

### ISSUE THREAD

**Issue by derek-ho** [opensearch-project/data-prepper#2456]
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. Ex. It would be nice to have [...]

**Describe the solution you'd like**

**Describe alternatives you've considered (Optional)**

**Additional context**

**Comment by dlvenable** [comment]
@derek-ho , I think supporting exemplars would be a great feature. Do you have a specific goal in mind? Or an expected outcome from Data Prepper?

**Comment by dlvenable** [comment]
Yes, I think that including exemplars in the model to send to OpenSearch makes sense.

Data Prepper has the ability to generate metrics from traces. And it has the ability to receive OTel metrics via gRPC. Are you expecting to have exemplars for both scenarios? Or a specific scenario?

**Comment by derek-ho** [comment]
@dlvenable I talked briefly with Krishna about this, he suggested of the following types: 

1. metrics passed through DataPrepper 
2. metrics generated by DataPrepper

To start with the former as the latter uses the former. I think I will start with that, thoughts?
